sdk/src/client/api/block_builder/input_selection/core/transition.rs

```diff
@@ -35,12 +35,22 @@ impl InputSelection {
         }
 
         // Do not create an alias output if it already exists.
-        if self
+        if let Some(alias_output) = self
             .outputs
             .iter()
-            .any(|output| is_alias_with_id_non_null(output, &alias_id))
+            .find(|output| is_alias_with_id_non_null(output, &alias_id))
         {
             log::debug!("No transition of {output_id:?}/{alias_id:?} as output already exists");
+
+            if (alias_output.as_alias().state_metadata() != input.state_metadata())
+                && (alias_output.as_alias().state_index() == input.state_index())
```

Comment:
It's not only `state_index` though, right? It's also amount, native_tokens and foundry_counter. And we should also check state transition.

Reply:
Which makes me thing we could use `StateTransitionVerifier::transition` if we manage to extract the context-related parts

Reply:
Yeah but it's very arbitrary to check `state_metadata` only :/

Reply:
I think I found a way that's okay now (I know some loop rounds can be reduced), if you think it's fine this way I would extend the transition validation it to nft and foundry outputs